src/mecca/platform/os/darwin/package.d

```diff
@@ -0,0 +1,158 @@
+    SIGKILL = 9, /// kill (cannot be caught or ignored)
+    SIGBUS = 10, /// bus error
+    SIGSEGV = 11, /// segmentation violation
+    SIGSYS = 12, /// bad argument to system call
+    SIGPIPE = 13, /// write on a pipe with no one to read it
+    SIGALRM = 14, /// alarm clock
+    SIGTERM = 15, /// software termination signal from kill
+    SIGURG = 16, /// urgent condition on IO channel
+    SIGSTOP = 17, /// sendable stop signal not from tty
+    SIGTSTP = 18, /// stop signal from tty
+    SIGCONT = 19, /// continue a stopped process
+    SIGCHLD = 20, /// to parent on child stop or exit
+    SIGTTIN = 21, /// to readers pgrp upon background tty read
+    SIGTTOU = 22, /// like TTIN for output if (tp->t_local&LTOSTOP)
+    SIGIO = 23, /// input/output possible signal
+    SIGXCPU = 24, /// exceeded CPU time limit
+    SIGXFSZ = 25, /// exceeded file size limit
+    SIGVTALRM = 26, /// virtual time alarm
+    SIGPROF = 27, /// profiling time alarm
+    SIGWINCH = 28, /// window size changes
+    SIGINFO = 29, /// information request
+    SIGUSR1 = 30, /// user defined signal 1
+    SIGUSR2 = 31 /// user defined signal 2
+}
+
+/**
+ * Represents the ID of a thread.
+ *
+ * This type is platform dependent.
+ */
+alias ThreadId = ulong;
+
+__gshared static immutable BLOCKED_SIGNALS = [
+    OSSignal.SIGHUP, OSSignal.SIGINT, OSSignal.SIGQUIT,
+    //OSSignal.SIGILL, OSSignal.SIGTRAP, OSSignal.SIGABRT,
+    //OSSignal.SIGBUS, OSSignal.SIGFPE, OSSignal.SIGKILL,
+    //OSSignal.SIGUSR1, OSSignal.SIGSEGV, OSSignal.SIGUSR2,
+    OSSignal.SIGPIPE, OSSignal.SIGALRM, OSSignal.SIGTERM,
+    //OSSignal.SIGSTKFLT, OSSignal.SIGCONT, OSSignal.SIGSTOP,
+    OSSignal.SIGCHLD, OSSignal.SIGTSTP, OSSignal.SIGTTIN,
+    OSSignal.SIGTTOU, OSSignal.SIGURG, OSSignal.SIGXCPU,
+    OSSignal.SIGXFSZ, OSSignal.SIGVTALRM, OSSignal.SIGPROF,
+    OSSignal.SIGWINCH, OSSignal.SIGIO,
+    //OSSignal.SIGSYS,
+];
+
+extern (C) private int pthread_threadid_np(pthread_t, ulong*) nothrow;
+
+/// Returns: the current thread ID
+ThreadId currentThreadId() @system nothrow
+{
+    import mecca.lib.exception : ASSERT;
+
+    enum assertMessage = "pthread_threadid_np failed, should not happen";
+
+    ulong threadId;
+    ASSERT!"assertMessage"(pthread_threadid_np(null, &threadId) == 0);
+
+    return threadId;
+}
+
+enum O_CLOEXEC = 0x1000000;
+enum F_DUPFD_CLOEXEC = 67;
+
+// this does not exist on Darwin
+enum EREMOTEIO = -1;
+
+// `pipe2` does not exist on Darwin so we're emulating it instead. This is
+// emulated by first calling the regular `pipe` followed by `fcntl` on the two
+// file descriptors. This is not thread safe.
+extern(C) private int pipe2(ref int[2] pipefd, int flags) nothrow @trusted @nogc
+{
+    import core.sys.posix.unistd : pipe;
+    import core.sys.posix.fcntl : fcntl, F_SETFD;
+
+    const pipeResult = pipe(pipefd);
+
+    if (pipeResult != 0)
+        return pipeResult;
+
+    foreach (fd; pipefd)
+    {
+        const result = fcntl(fd, F_SETFD, flags);
```

Comment:
Probably best to `F_GETFD` first and bitwise-or the flags in. Can extract this to a function to enable flags.

Reply:
Fixed, I hope.